chitchat/src/lib.rs

```diff
@@ -47,71 +47,52 @@ const MAX_UDP_DATAGRAM_PAYLOAD_SIZE: usize = 65_507;
 
 pub type Version = u64;
 
-/// [`NodeId`] represents a Chitchat Node identifier.
+/// For the lifetime of a cluster, nodes can go down and come back up multiple times. They may also
```

Comment:
Really nice docs, it's shorter and more accurate, I like to be explicit with the `join_id`.

Nitpick: you can ignore but I have a small preference for generation ID, how did you come up with `join_id` (google? :))?

Reply:
Generation ID is good too. I'll go for that.